backend/src/services/user.go

```diff
@@ -3,19 +3,57 @@ package services
 import (
 	"github.com/GenerateNU/sac/backend/src/models"
 	"github.com/GenerateNU/sac/backend/src/transactions"
-
+	"github.com/GenerateNU/sac/backend/src/types"
+	"github.com/GenerateNU/sac/backend/src/utilities"
+	"github.com/go-playground/validator/v10"
+	"github.com/gofiber/fiber/v2"
 	"gorm.io/gorm"
 )
 
 type UserServiceInterface interface {
+	CreateUserFromParams(params types.UserParams) (models.User, error)
 	GetAllUsers() ([]models.User, error)
+	UpdateUser(id string, params types.UserParams) (models.User, error)
 }
 
 type UserService struct {
 	DB *gorm.DB
 }
 
+// Creates a models.User from params. This *does not* interact with the database at all; the value will need to be
+// passed to gorm.Db.Create(interface{}) for it to be persisted.
+func (u *UserService) CreateUserFromParams(params types.UserParams) (models.User, error) {
```

Comment:
@DOOduneye another repo organization Q, where should we put this? do we even need this with this with the updated validation in models.User?